### DISCUSSION TITLE
Do we need executor after merge ?

### DISCUSSION THREAD

**Discussion by wiryonolau** [OffchainLabs/prysm#11446] [Q&A] [answered]
Hi do we need executor after successful merge for solo validatator ?
executor (geth) is not eth2 correct ?

**Comment by prestonvanloon** [comment, marked as answer]
Yes, you need an execution client after the merge. 

Please see our documentation here: https://docs.prylabs.network/docs/prepare-for-merge

**Reply by wiryonolau** [reply]
Can we run geth using syncmode light for execution layer ?

**Reply by prestonvanloon** [reply]
No, you will need a full node for this. Pruning is OK, but you must have full state access. Light syncmode will not be enough.

**Reply by wiryonolau** [reply]
so only **full** can be use not even **snap** syncmode ?

**Reply by prestonvanloon** [reply]
snap sync is OK. You just need the full state (not a light client)

**Reply by wiryonolau** [reply]
Thanks for the info
